tests/e2e/gc_rwx_reclaim_policy.go

```diff
@@ -0,0 +1,518 @@
+		14. Delete Pod1
+		15. Verify CnsFileAccessConfig CRD is deleted
+		16. Verify if the Pod is successfully deleted
+		17. Delete PVC in GC
+		18. Verify if PVC and PV still persists in the SV cluster
+		19. Verify PVC name is removed from PV entry on CNS
+		20. Create a PV with reclaim policy=delete & access mode ReadWriteMany in GC
+		    using the bound PVC in the SV cluster as the volume id PV1
+		21. Create PVC2 with "ReadWriteMany" access mode which gets bounds to PV2 (reclaim policy=delete)
+		22. Verify CnsVolumeMetadata CRD is created
+		23. Verify health status of PVC
+		24. Verify volume is created on CNS by using CNSQuery API and also check metadata is pushed to CNS
+		25. Create Pod2 using PVC2 created above
+		26. Verify CnsFileAccessConfig CRD is created
+		27. Verify Pod2 is in the Running phase
+		28. Verify ACL net permission set by calling CNSQuery for the file volume
+		29. Read the file (file1.txt) created by Pod1 from Pod2. Check if reading is successful
+		30. Create a new file (file2.txt) at the mount path. Check if the creation is successful
+		31. Delete Pod2
+		32. Verify CnsFileAccessConfig CRD is deleted
+		33. Verify if Pod2 is successfully deleted
+		34. Delete PVC2 and PVs from GC
+		35. Verify if PVCs and PVs are deleted in the SV cluster and GC
+		36. Verify CnsVolumeMetadata CRD is deleted
+		37. Check if the VolumeID is deleted from CNS by using CNSQuery API
+	*/
+	ginkgo.It("Verify Reclaim Policy Retain with file volume", func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		var storageclasspvc *storagev1.StorageClass
+		var pvclaim *v1.PersistentVolumeClaim
+		var err error
+
+		ginkgo.By("CNS_TEST: Running for GC setup")
+		scParameters[svStorageClassName] = storagePolicyName
+		ginkgo.By("Creating a PVC")
+		storageclasspvc, pvclaim, err = createPVCAndStorageClass(client,
+			namespace, nil, scParameters, diskSize, nil, "", false, v1.ReadWriteMany)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		defer func() {
+			err = client.StorageV1().StorageClasses().Delete(ctx, storageclasspvc.Name, *metav1.NewDeleteOptions(0))
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		ginkgo.By("Expect claim to provision volume successfully")
+		persistentvolumes, err := fpv.WaitForPVClaimBoundPhase(client,
+			[]*v1.PersistentVolumeClaim{pvclaim}, framework.ClaimProvisionTimeout)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred(), "Failed to provision volume")
+
+		volHandle := persistentvolumes[0].Spec.CSI.VolumeHandle
+		gomega.Expect(volHandle).NotTo(gomega.BeEmpty())
+		volumeID := getVolumeIDFromSupervisorCluster(volHandle)
+		gomega.Expect(volumeID).NotTo(gomega.BeEmpty())
+
+		defer func() {
+			err = fpv.DeletePersistentVolumeClaim(client, pvclaim.Name, pvclaim.Namespace)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			err = e2eVSphere.waitForCNSVolumeToBeDeleted(volHandle)
```

Comment:
you would have to delete PV and CNS volume here since your using `Retain` reclaim policy and then verify cns volume is not there

Reply:
Fixed